mountpoint-s3-client/src/endpoint.rs

```diff
@@ -1,109 +1,80 @@
-use lazy_static::lazy_static;
 use mountpoint_s3_crt::common::allocator::Allocator;
 use mountpoint_s3_crt::common::uri::Uri;
 use mountpoint_s3_crt::s3::endpoint_resolver::{RequestContext, ResolverError, RuleEngine};
-use regex::Regex;
 use std::ffi::OsStr;
 use std::os::unix::prelude::OsStrExt;
 use thiserror::Error;
 
-lazy_static! {
-    /// Bucket names that are acceptable as virtual host names for DNS
-    static ref VALID_DNS_REGEX: Regex = Regex::new(r"[a-z0-9][a-z0-9\-]*[a-z0-9]").unwrap();
-}
+use crate::EndpointConfig;
 
 #[derive(Debug, Clone)]
 #[non_exhaustive]
 pub struct Endpoint {
```

Comment:
My rough thinking was that this struct doesn't need to exist any more, and instead things work something like this:
- the `S3CrtClient` has an `endpoint_config: EndpointConfig` field
- in `S3CrtClient::new_request_template`, the first line:
    ```rust
    let (uri, path_prefix) = self.endpoint.for_bucket(bucket)?;
    ```
    is now something like
    ```rust
    let (uri, path_prefix) = self.endpoint_config.resolve_for_bucket(bucket)?;
    ````
    and that `resolve_for_bucket` function is what invokes the endpoint resolver.
    - I don't know exactly how `ForcePathStyle` works with the endpoint resolver, but something needs to arrange for `path_prefix` to be set correctly so it can be used later. Or maybe the path prefix is already in the `Uri` and `S3Message::set_request_path_and_query` can take care of setting the request up correctly?

Reply:
Looking into how path_prefix works with the resolver.